mautic/mautic
# Fix the campaign condition 'is a member of segment'

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


The campaign condition "Is a member of the segment" is not working properly sometimes. This PR fixes that.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create two new contact or use existing ones.
3. Create the first segment and add the contacts here.
4. Create the second segment with any filter and add the same contacts to the segment by filter.
5. Remove one of these contacts from the segment manually (e.g. via the Contact Preference Center).
6. Create two new tags. E.g., "segment member" and "not a segment member".
7. Create a new campaign.
8. Select the first segment as "Contact Source".
9. Add a new condition "Is a member of the segment" and select the second segment.
10. Add actions "Modify contact's tags" to check the condition "Is a member of the segment". If the condition is true, add the tag "segment member". If not, add the tag "not a segment member".
11. The condition works properly now. Before, the false branch didn't work.
![image](https://user-images.githubusercontent.com/96085911/207561610-1dc2d7d9-6128-46e5-84ee-1b66c293a20e.png)
![image](https://user-images.githubusercontent.com/96085911/207561046-cedd295f-82e0-4b28-b1ba-267a6642781d.png)
![image](https://user-images.githubusercontent.com/96085911/207561170-4918e15b-a6db-4c81-8b6d-d5a536b3d421.png)
